Neos.Flow/Classes/Reflection/ParameterReflection.php

```diff
@@ -57,9 +57,9 @@ public function getClass()
     public function getBuiltinType()
     {
         $type = $this->getType();
-        if ($type === null || !$type->isBuiltin()) {
+        if (!$type instanceof \ReflectionNamedType) {
             return null;
         }
-        return $type instanceof \ReflectionNamedType ? $type->getName() : (string)$type;
+        return $type->isBuiltin() ? $type->getName() : null;
```

Comment:
According to https://www.php.net/manual/en/migration80.incompatible.php#migration80.incompatible.reflection
> ReflectionType::isBuiltin() method has been moved to ReflectionNamedType. ReflectionUnionType does not have it.

Not sure if the change addresses this correctly!

Reply:
I think we still need to do the instanceof check, because else we lose compatibility with PHP 7.x. Also, we did not invoke `isBuiltin()` here at all?

Reply:
the instanceof check is in line 60 now but I'm not sure if 61 should be `return (string)$type` instead

Reply:
In PHP8 there are only `ReflectionNamedType` and `ReflectionUnionType`. Hence the question becomes: How should we handle a union of a builtin type with other types here?
For PHP >= 7.1 only `ReflectionNamedType` exists, so returning `null` should be okay. If we want to be super sure, it should be `PHP < 8 ? (string)$type : null`